### PR TITLE
schroot: init at 1.6.13-5

### PR DESCRIPTION
**Pull request by bjsowa**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2411.section.md [hunk 1]
```diff
@@ -75,6 +75,8 @@
 
 - [Cyrus IMAP](https://github.com/cyrusimap/cyrus-imapd), an email, contacts and calendar server. Available as [services.cyrus-imap](#opt-services.cyrus-imap.enable) service.
 
+- [Schroot](), a lightweight virtualisation tool. Securely enter a chroot and run a command or login shell. Available as [programs.schroot](#opt-programs.schroot.enable).
```
Comment (FliegendeWurst) [inline nixos/doc/manual/release-notes/rl-2411.section.md:78]: The link leads nowhere

File: pkgs/by-name/sc/schroot/debian-patches.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Generated by debian-patches.sh from debian-patches.txt
+let
+  prefix = "https://sources.debian.org/data/main/s/schroot/1.6.13-5/debian/patches";
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/schroot/debian-patches.nix:3]: I think this download will eventually break (as soon as the version is updated at Debian).

File: pkgs/by-name/sc/schroot/package.nix [hunk 3]
```diff
@@ -0,0 +1,89 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  pkgs,
+  buildPackages,
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/schroot/package.nix:6]: Should use individual packages.

File: pkgs/by-name/sc/schroot/package.nix [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  ];
+
+  buildInputs = with pkgs; [
+    boost
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_INSTALL_LOCALSTATEDIR=/var"
+    "-DSCHROOT_SYSCONF_DIR=/etc/schroot"
+    "-DSCHROOT_CONF_SETUP_D=${placeholder "out"}/etc/schroot/setup.d"
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/schroot/package.nix:54]: There are some functions in lib to generate these, like lib.cmakeFeature

File: pkgs/by-name/sc/schroot/package.nix [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    description = "Lightweight virtualisation tool";
+    longDescription = ''
+      Schroot is a program that allows the user to run a command or a login shell in a chroot environment.
+    '';
+    homepage = "https://codeberg.org/shelter/reschroot";
+    changelog = "https://codeberg.org/shelter/reschroot/raw/tag/release/reschroot-${upstream-version}/NEWS";
+    mainProgram = "schroot";
+    maintainers = with lib.maintainers; [ bjsowa ];
+    license = with licenses; [ gpl3 ];
+    platforms = lib.platforms.linux;
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/schroot/package.nix:87]:
```suggestion
  meta = {
    description = "Lightweight virtualisation tool";
    longDescription = ''
      Schroot is a program that allows the user to run a command or a login shell in a chroot environment.
    '';
    homepage = "https://codeberg.org/shelter/reschroot";
    changelog = "https://codeberg.org/shelter/reschroot/raw/tag/release/reschroot-${upstream-version}/NEWS";
    mainProgram = "schroot";
    maintainers = with lib.maintainers; [ bjsowa ];
    license = lib.licenses.gpl3Plus;
    platforms = lib.platforms.linux;
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sc/schroot/debian-patches.nix (hunk 2 above)
Comment (bjsowa) [reply pkgs/by-name/sc/schroot/debian-patches.nix:3]: Would it be ok if I used salsa.debian.org instead? i. e. https://salsa.debian.org/debian/schroot/-/raw/debian/master/1.6.13-4/debian/patches

Comment (FliegendeWurst) [reply pkgs/by-name/sc/schroot/debian-patches.nix:3]: If they are accessible without logging in, yes

File: pkgs/by-name/sc/schroot/debian-patches.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
+# Generated by debian-patches.sh from debian-patches.txt
+let
+  prefix = "https://salsa.debian.org/debian/schroot/-/raw/debian/master/1.6.13-4/debian/patches";
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/schroot/debian-patches.nix:3]: This should use a specific commit instead of `master`

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sc/schroot/debian-patches.nix (hunk 6 above)
Comment (bjsowa) [reply pkgs/by-name/sc/schroot/debian-patches.nix:3]: It is using the tag `debian/master/1.6.13-4` (it's not a branch name)

Comment (FliegendeWurst) [reply pkgs/by-name/sc/schroot/debian-patches.nix:3]: Interesting naming scheme. Then it is okay.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by bjsowa** [comment]
Fixed merge conflict by moving the release note to 25.05

**Review by FliegendeWurst — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
So I was just testing this PR on staging too.. and over there we have a new default boost version, causing the build to fail
```
/build/reschroot/sbuild/sbuild-chroot-config.cc: In member function 'void sbuild::chroot_config::add_config_directory(const std::string&, const std::string&)':
/build/reschroot/sbuild/sbuild-chroot-config.cc:152:22: error: 'directory_iterator' is not a member of 'boost::filesystem'; did you mean 'directory_entry'?
  152 |   boost::filesystem::directory_iterator end_iter;
      |                      ^~~~~~~~~~~~~~~~~~
      |                      directory_entry
/build/reschroot/sbuild/sbuild-chroot-config.cc:153:27: error: 'directory_iterator' is not a member of 'boost::filesystem'; did you mean 'directory_entry'?
  153 |   for (boost::filesystem::directory_iterator dirent(dirpath);
      |                           ^~~~~~~~~~~~~~~~~~
      |                           directory_entry
/build/reschroot/sbuild/sbuild-chroot-config.cc:154:8: error: 'dirent' was not declared in this scope
  154 |        dirent != end_iter;
      |        ^~~~~~
/build/reschroot/sbuild/sbuild-chroot-config.cc:154:18: error: 'end_iter' was not declared in this scope
  154 |        dirent != end_iter;
      |                  ^~~~~~~~
/build/reschroot/sbuild/sbuild-run-parts.cc: In constructor 'sbuild::run_parts::run_parts(const std::string&, bool, bool, mode_t)':
/build/reschroot/sbuild/sbuild-run-parts.cc:79:22: error: 'directory_iterator' is not a member of 'boost::filesystem'; did you mean 'directory_entry'?
   79 |   boost::filesystem::directory_iterator end_iter;
      |                      ^~~~~~~~~~~~~~~~~~
      |                      directory_entry
/build/reschroot/sbuild/sbuild-run-parts.cc:80:27: error: 'directory_iterator' is not a member of 'boost::filesystem'; did you mean 'directory_entry'?
   80 |   for (boost::filesystem::directory_iterator dirent(dirpath);
      |                           ^~~~~~~~~~~~~~~~~~
      |                           directory_entry
```
It appears the file is missing an include for `boost/filesystem/directory.hpp`

**Comment by bjsowa** [comment]
@FliegendeWurst I fixed the boost includes, this should now build on staging

File: pkgs/by-name/sc/schroot/package.nix [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    # The wrapped script are put into a separate directory to not be run by schroot during setup
+    mkdir $out/etc/schroot/setup.d.wrapped
+    cd $out/etc/schroot/setup.d
+    find * -type f | while read -r file; do
+      mv "$file" $out/etc/schroot/setup.d.wrapped
+      makeWrapper "$out/etc/schroot/setup.d.wrapped/$file" "$file" --set PATH ${scripts-bin-path}
+    done
+
+    # Get rid of stuff that's (probably) not needed
+    rm -vrf $out/lib $out/include
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/schroot/package.nix:90]: This is run after install, so it has no effect

Comment (FliegendeWurst) [reply pkgs/by-name/sc/schroot/package.nix:90]: They only contain files for "sbuild" so they are probably safe to remove.

**Comment by bjsowa** [comment]
@wegank Why was the `approvals: 1` label removed?

**Comment by FliegendeWurst** [comment]
It is an automatic script. Label is removed if PR was changed after the approval.

**Pull request merged**